awswrangler/timestream/_read.py

```diff
@@ -155,3 +167,272 @@ def query(
+    pyarrow_additional_kwargs : Dict[str, Any], optional
+        Forwarded to `to_pandas` method converting from PyArrow tables to Pandas DataFrame.
+        Valid values include "split_blocks", "self_destruct", "ignore_metadata".
+        e.g. pyarrow_additional_kwargs={'split_blocks': True}.
+
+    Returns
+    -------
+    Union[pandas.DataFrame, Iterator[pandas.DataFrame]]
+        Result as Pandas DataFrame(s).
+
+    Examples
+    --------
+    Unload and read as Parquet (default).
+
+    >>> import awswrangler as wr
+    >>> df = wr.timestream.unload(
+    ...     sql="SELECT time, measure, dimension FROM database.mytable",
+    ...     path="s3://bucket/extracted_parquet_files/",
+    ... )
+
+    Unload and read partitioned Parquet. Note: partition columns must be at the end of the table.
+
+    >>> import awswrangler as wr
+    >>> df = wr.timestream.unload(
+    ...     sql="SELECT time, measure, dim1, dim2 FROM database.mytable",
+    ...     path="s3://bucket/extracted_parquet_files/",
+    ...     partition_cols=["dim2"],
+    ... )
+
+    Unload and read as CSV.
+
+    >>> import awswrangler as wr
+    >>> df = wr.timestream.unload(
+    ...     sql="SELECT time, measure, dimension FROM database.mytable",
+    ...     path="s3://bucket/extracted_parquet_files/",
+    ...     unload_format="CSV",
+    ... )
+
+    """
+    path = path if path.endswith("/") else f"{path}/"
+
+    if unload_format not in [None, "CSV", "PARQUET"]:
+        raise exceptions.InvalidArgumentValue("<unload_format> argument must be 'CSV' or 'PARQUET'")
+
+    unload_to_files(
+        sql=sql,
+        path=path,
+        unload_format=unload_format,
+        compression=compression,
+        partition_cols=partition_cols,
+        encryption=encryption,
+        kms_key_id=kms_key_id,
+        field_delimiter=field_delimiter,
+        escaped_by=escaped_by,
+        boto3_session=boto3_session,
+    )
+    results_path = f"{path}results/"
+    if unload_format == "CSV":
+        column_names: List[str] = _get_column_names_from_metadata(path, boto3_session)
+        df: Union[pd.DataFrame, Iterator[pd.DataFrame]] = s3.read_csv(
+            path=results_path,
+            header=None,
+            names=[column for column in column_names if column not in set(partition_cols)]
+            if partition_cols is not None
+            else column_names,
+            dataset=True,
+            use_threads=use_threads,
+            boto3_session=boto3_session,
+            s3_additional_kwargs=s3_additional_kwargs,
+        )
+    else:
+        df: Union[pd.DataFrame, Iterator[pd.DataFrame]] = s3.read_parquet(
+            path=results_path,
+            chunked=chunked,
+            dataset=True,
+            use_threads=use_threads,
+            boto3_session=boto3_session,
+            s3_additional_kwargs=s3_additional_kwargs,
+            pyarrow_additional_kwargs=pyarrow_additional_kwargs,
+        )
+
+    if keep_files is False:
```

Comment:
Any reason why we are not wrapping the keep files within a try/expect/finally?

Reply:
Good call, thanks